src/GeoJsonPages/GeoJsonContentHandler.php

```diff
@@ -34,19 +38,31 @@ protected function fillParserOutput(
 		ParserOutput &$parserOutput
 	) {
 		'@phan-var GeoJsonContent $content';
+		
+		// @see JsonContentHandler -> fillParserOutput
+		if ( $cpoParams->getGenerateHtml() ) {
+			if ( $content->isValid() ) {
+				$text = $content->getData()->getValue();
+				$parserOutput->setText( $content->rootValueTable( $content->getData()->getValue() ) );
+				
+			} else {
+				$error = wfMessage( 'invalid-json-data' )->parse();
+				$parserOutput->setText( $error );
+			}
```

Comment:
$parserOutput->setText( newMethod( args ) )

Reply:
@thomas-topway-it you added a bunch of random details that are now needed for compat reasons. It's best to encapsulate those in a dedicated function, to not pollute the main method.

Reply:
https://www.entropywins.wtf/blog/2018/10/30/readable-functions-do-one-thing/

Reply:
fixed, now using the parent fillParserOutput and added additional condition